File: projects/06/ArrayProcessing.rb
# ArrayProcessing.rb creates a string of numbers from the user and prints them
# Precondition: size input is greater than or equal to 0
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Jacob Brink
# Date: 3/20/2020
# Project: 06
########################################################

###############################################
# fillarray() fills the array with user input
# Receive: anArray, an array of numbers
# Return: array populated with user input
################################################
def fillArray (arraySize)
  array = Array.new(arraySize)
  for i in 1..arraySize
    print "Enter element #{i}: "
    array[i] = gets.to_f
    puts "\n"
  end  
end

################################################
# process() prints the elements of the given array
# Receive: an array of numeric values 
# Return: void
######################################################
def process (array)
  array.each { |elem| print "#{elem}\n" }
end


if __FILE__ == $0
  print "Enter size: ";
  size = gets.to_i
  puts "\n";
  process(fillArray(size));
end

File: projects/06/clojure/src/arrayprocessing.clj
;; arrayprocessing.clj "test-drives" function arrayprocessing() and arrayinput().
;; precondition: size given is nonnegative
;; Output: the arrayprocessing of a sequence of numbers.
;;
;; Usage: clojure -m arrayprocessing
;;
;; Begun by: Dr. Adams, CS 214 at Calvin College.
;; Completed by: Jacob Brink
;; Date: 3/19/2020
;; Project: 06
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns arrayprocessing)



;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;; arrayprocessing() prints the values of the given array
;; Receive: aVec, a vector of numbers. 
;; Return: the arrayprocessing of the numbers in aVec.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
   
(defn arrayprocessing [array]
  (loop [i 0]
    (if (< i (count array))
      (do
        (printf "%.3f\n" (get array i))
        (recur (inc i))
        )
      )
    )
  )



;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;; arrayinput inputs the values in a vector.       
;; Receive: arraySize an integer
;; Return: array filled with user input
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn arrayinput [arraySize]
  (loop [i 0 array []]
    (if (< i arraySize)
      (do (print "Enter value: ") (flush)
          (let [userInput (float (read))]
            (recur (inc i) (conj array userInput))
         )
          )
      array
      )
    )
  )

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;; main function to test functions sum() and arrayprocessing()
;; Output: the sum and arrayprocessing of some test vectors.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
  (print "\nEnter Size: ") (flush)

  (let [ arraySize (int (read)) ]
    (arrayprocessing (arrayinput arraySize))
    )

)

